packages/frontend/src/components/ErrorResult/index.tsx

```diff
@@ -0,0 +1,30 @@
+import { IJSONObject, IStepError } from '@plumber/types'
+
+import GenericErrorResult from './GenericErrorResult'
+import SpecificErrorResult from './SpecificErrorResult'
+
+const isStepError = (value: IStepError | IJSONObject): value is IStepError => {
+  // Narrowing type for IStepError using type predicate
+  return (
+    value &&
+    typeof value === 'object' &&
+    !!value.name &&
+    !!value.solution &&
+    !!value.position &&
+    !!value.appName
+  )
+}
```

Comment:
nit: consider locating the type guard with the type (or interface) declaration since they are related code